lib/debugger/src/dbg_ieval.erl

```diff
@@ -1807,6 +1985,34 @@ merge_bindings([{Name,V}|B1s], B2s, Ieval) ->
 merge_bindings([], B2s, _Ieval) ->
     B2s.
 
+zip_add_bindings(Bs1, Bs2) when is_map(Bs1), is_map(Bs2) ->
+    lists:foldl(fun(Key, Acc) ->
+                        case maps:find(Key, Bs1) of
+                            {ok, Val} ->
+                                case maps:find(Key, Bs2) of
+                                    {ok, Val} ->
+                                        Acc;
+                                    {ok, _Value} ->
+                                        nomatch
```

Comment:
This looks a bit odd, `Acc` will be `nomatch` on the next go around, which may cause the `maps:put/3` below to fail. Looking at the usage, as well as the `orddict` version below, I'm guessing that you want iteration to bail out and the entire thing to return `nomatch`. How about something like:

```erlang
%% Completely untested
zip_add_bindings(LHS, RHS) when is_map(LHS), is_map(RHS) ->
    zab_map(maps:keys(RHS), LHS, RHS);
zip_add_bindings(LHS, RHS) when is_list(LHS), is_list(RHS) ->
    zab_orddict(orddict:to_list(LHS), RHS).

zab_map([Key | Keys], LHS, RHS) ->
    case {LHS, RHS} of
        {#{ Key := Same }, #{ Key := Same }} -> zab_map(Keys, LHS, RHS);
        {#{ Key := _ }, #{}} -> nomatch;
        {#{}, #{ Key := Value }} -> zab_map(Keys, LHS, RHS#{ Key => Value });
    end;
zab_map([], RHS) ->
    RHS.

zab_orddict([{Name, Value} | KVPs], RHS) ->
    case orddict:find(Name, RHS) of
        {ok, Value} -> zab_orddict(KVPs, RHS);
        {ok, _Other} -> nomatch;
        error -> zab_orddict(KVPs, orddict:store(Name, Val, Acc))
    end;
zab_orddict([], RHS) ->
    RHS.
```

(`erl_eval` seems to have the same issue)

Reply:
Yes, once `zip_add_bindings` finds discrepancies, it should return `nomatch` instead of continuing. I'll fix it.